pkg/cableengine/syncer/syncer.go

```diff
@@ -0,0 +1,178 @@
+package syncer
+
+import (
+	"fmt"
+	"reflect"
+	"strconv"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+
+	v1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/cableengine"
+	v1typed "github.com/submariner-io/submariner/pkg/client/clientset/versioned/typed/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/log"
+)
+
+type GatewaySyncer struct {
+	sync.Mutex
```

Comment:
This mutex is not really needed as no local state is mutated and `syncGatewayStatus` is not called concurrently.

Reply:
I'm not sure if the GatewayInterface client can be called concurrently, are we sure about that? that was the reason.

Reply:
the clientset? yeah they're  thread-safe - they carry no state - they're just front-ends to the API server.

Reply:
Done.